master/master/master_server.py

```diff
@@ -0,0 +1,69 @@
+import requests
+from flask import Flask, request, jsonify
+from master.storage import save_to_persistent_storage, load_from_persistent_storage
+
+app = Flask(__name__)
+
+class MasterServer:
+    def __init__(self):
+        self.database = load_from_persistent_storage()
+        self.edge_nodes = []
+
+    def save_database(self):
+        save_to_persistent_storage(self.database)
+
+    def set_value(self, key, value):
+        self.database[key] = value
+        self.broadcast_set(key, value)
+        self.save_database()
+        return {"status": "success", "key": key, "value": value}
+
+    def get_value(self, key):
+        value = self.database.get(key, None)
+        return {"status": "success", "key": key, "value": value}
+
+    def broadcast_set(self, key, value):
+        for node in self.edge_nodes:
+            try:
+                requests.post(f"{node['url']}/keys/{key}", json={"value": value})
```

Comment:
Sequential blocking request for broadcasting isn't appropriate, better use grequests/write some multithreded code to send all requests at once